Zend/zend_API.c

```diff
@@ -252,6 +252,26 @@ ZEND_API ZEND_COLD void ZEND_FASTCALL zend_wrong_parameter_class_or_null_error(i
 }
 /* }}} */
 
+ZEND_API ZEND_COLD void ZEND_FASTCALL zend_wrong_parameter_string_or_class_error(int num, const char *name, zval *arg) /* {{{ */
+{
+	if (EG(exception)) {
+		return;
+	}
+
+	zend_argument_type_error(num, "must be of type string|%s, %s given", name, zend_zval_type_name(arg));
```

Comment:
I believe the canonical type ordering is `%s|string`.

Reply:
What about the string|object case? Or that's okay, because `object` is an internal type?

Reply:
> $ sapi/cli/php -r '(function(string|object $x){})([]);'
> 
> Fatal error: Uncaught TypeError: {closure}(): Argument #1 ($x) must be of type object|string, array given

It's `object|string` as well.

Reply:
I've just adapted the argument types to follow this order, that's why the new commit.